core/src/main/java/com/graphhopper/routing/weighting/AvoidWeighting.java

```diff
@@ -0,0 +1,114 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.routing.weighting;
+
+import com.graphhopper.routing.profiles.EnumEncodedValue;
+import com.graphhopper.routing.profiles.RoadClass;
+import com.graphhopper.routing.profiles.RoadEnvironment;
+import com.graphhopper.routing.profiles.Toll;
+import com.graphhopper.routing.util.FlagEncoder;
+import com.graphhopper.util.EdgeIteratorState;
+import com.graphhopper.util.PMap;
+
+import static com.graphhopper.routing.profiles.RoadClass.*;
+import static com.graphhopper.routing.profiles.RoadEnvironment.*;
+
+/**
+ * A simple Weighting to demonstrate how to avoid certain road classes based on the EncodedValue RoadClass.
+ * <p>
+ * Note: might be replaced in future versions.
+ */
+public class AvoidWeighting extends ShortFastestWeighting {
+
+    private boolean avoidFerry;
+    private boolean avoidBridge;
+    private boolean avoidFord;
+    private boolean avoidTunnel;
+
+    private boolean avoidToll;
+
+    private boolean avoidMotorway;
+    private boolean avoidTrunk;
+    private boolean avoidPrimary;
+    private boolean avoidSecondary;
+    private boolean avoidTertiary;
+    private boolean avoidTrack;
+    private boolean avoidResidential;
+    private EnumEncodedValue<Toll> tollEnc;
+    private EnumEncodedValue<RoadEnvironment> roadEnvEnc;
+    private EnumEncodedValue<RoadClass> roadClassEnc;
+    private double avoidFactor;
+
+    public AvoidWeighting(FlagEncoder encoder, PMap map) {
+        super(encoder, map);
+
+        String avoidStr = map.get("avoid", "");
+        if (encoder.hasEncodedValue(RoadEnvironment.KEY)) {
+            avoidFerry = avoidStr.contains("ferry");
+            avoidBridge = avoidStr.contains("bridge");
+            avoidFord = avoidStr.contains("ford");
+            avoidTunnel = avoidStr.contains("tunnel");
+            roadEnvEnc = encoder.getEnumEncodedValue(RoadEnvironment.KEY, RoadEnvironment.class);
+        }
+
+        if (encoder.hasEncodedValue(Toll.KEY)) {
+            avoidToll = avoidStr.contains("toll");
+            tollEnc = encoder.getEnumEncodedValue(Toll.KEY, Toll.class);
+        }
+
+        if (encoder.hasEncodedValue(RoadClass.KEY)) {
+            avoidMotorway = avoidStr.contains("motorway");
+            avoidTrunk = avoidStr.contains("trunk");
+            avoidPrimary = avoidStr.contains("primary");
+            avoidSecondary = avoidStr.contains("secondary");
+            avoidTertiary = avoidStr.contains("tertiary");
+            avoidTrack = avoidStr.contains("track");
+            avoidResidential = avoidStr.contains("residential");
+            roadClassEnc = encoder.getEnumEncodedValue(RoadClass.KEY, RoadClass.class);
+        }
+
+        // can be used for preferring too
+        avoidFactor = Math.min(10, Math.max(0.1, map.getDouble("avoid.factor", 10)));
```

Comment:
Is/Should there be a safety check when using LM that the avoid.factor is at least 1?

Reply:
Maybe we move the AvoidWeighting stuff outside of this already big PR as this already introduces a feature that is maybe not wanted.

Reply:
Yes, I think this is a good idea :)